### ISSUE TITLE
DenseNet FCN

### ISSUE THREAD

**Issue by FuriouslyCurious** [pytorch/vision#131]
Hi @orashi and @gpleiss - thank you for adding DenseNet!  

I was wondering if you have looked at DenseNet Fully Convolutional paper which gives phenomenal scores for segmentation : https://arxiv.org/pdf/1611.09326v1.pdf

I will owe you all a round of beers if you can consider implementing DenseNet FCN too.

Great day,
FC

**Comment by gpleiss** [comment]
Haha who can say no to that 😄 

The [original implementation](https://github.com/SimJeg/FC-DenseNet) is in Theano. We could probably use the saved weights for that model, but it may be better to train from scratch.

It might make sense to hold off until the CamVid dataset is added (#90).

**Comment by felixgwu** [comment]
Hi, @FuriouslyCurious I have an implementation of FC-DenseNet at head already.
I can submit a PR is needed.
However, I couldn't match their reported mIoU and accuracy by training from scratch.
I've submitted an issue to their repo, but he is busy to help these days.

**Comment by FuriouslyCurious** [comment]
Thank you @gpleiss and @felixgwu 

I am planning to train a Dense FC model from scratch using medical data and publish the weights / trained model for medical research.  

@felixgwu if you share the code in a GIST I will love to try it out for some experiments I am running.

**Comment by felixgwu** [comment]
@FuriouslyCurious 
Here is the code of the model. Hopefully, it helps your research.
https://gist.github.com/felixgwu/045c887b6ccdf0edf4648da0c40bcc12

**Comment by FuriouslyCurious** [comment]
Thank you @felixgwu !

**Comment by ycszen** [comment]
Thank you! @FuriouslyCurious

**Comment by youngfly11** [comment]
How can i import  the saved weight in the original implementation in Theano to the pytorch  with the same architechture? @felixgwu 
As I know,  the saved weight is numpy format. and it seem difficult to initial the weight in pytorch one by one, because the net is so deep. Thank you for advanced

**Comment by EliasVansteenkiste** [comment]
@felixgwu Do you remember how far you where off from the original reported mIoU and accuracy by training from scratch?

**Comment by EliasVansteenkiste** [comment]
I found your results in [FC-DenseNet issue 11](https://github.com/SimJeg/FC-DenseNet/issues/11).
Did you manage to further improve the results?

**Comment by FuriouslyCurious** [comment]
@EliasVansteenkiste and @felixgwu Check out the Keras Tiramisu implementation below:  Developer @titu1994 used SubPixelConvolution instead of Deconvolution as default method for upsampling.  Not sure if that helps accuracy, but worth trying.

https://github.com/titu1994/Fully-Connected-DenseNets-Semantic-Segmentation

**Comment by alykhantejani** [comment]
If you want to use sub-pixel convolution for upscaling you can use the [`PixelShuffle`](http://pytorch.org/docs/master/nn.html?highlight=pixelshuffle#torch.nn.PixelShuffle) layer in PyTorch

**Comment by youngfly11** [comment]
@felixgwu .  Thank you for you code implementation of FC-DenseNet in pytorch. Recent, i want to reproduce the result about FC-DenseNet, but my loss does not converge at all.  would you share me you dataLoader so that i can compare it with my code? and if i initialize the learning rate with 1e-3, the loss will be infinity.  I am so puzzled with it .

**Comment by wisdomdeng** [comment]
@felixgwu Thanks for your code. Could you please check line85 of your code? I think the last transition block before upsampling should have only_new set to false.
